pkg/kn/commands/service/create_test.go

```diff
@@ -493,3 +494,23 @@ func TestServiceCreateWithServiceAccountName(t *testing.T) {
 		t.Fatalf("wrong service account name:%v", template.Spec.ServiceAccountName)
 	}
 }
+
+func TestServiceCreateWithClusterLocal(t *testing.T) {
+	action, created, _, err := fakeServiceCreate([]string{
+		"service", "create", "foo", "--image", "gcr.io/foo/bar:baz",
+		"--cluster-local"}, false, false)
+
+	if err != nil {
+		t.Fatal(err)
+	} else if !action.Matches("create", "services") {
+		t.Fatalf("Bad action %v", action)
+	}
+
+	template, err := servinglib.RevisionTemplateOfService(created)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, present := template.Labels[config.VisibilityLabelKey]
+	assert.Assert(t, !present)
```

Comment:
Could we please also check for the proper value of the annotation ? Maybe also in update_test.go that the annotation gets removed when using `--no-cluster-local` to complete the scenario.

Reply:
Actually, slight problem adding the `update_test.go`. Seems like removing that label causes an API error:

```bash
Internal error occurred: admission webhook "webhook.serving.knative.dev" denied the request: mutation failed: Saw the following changes without a name change (-old +new): spec.configuration.template
{*v1alpha1.RevisionTemplateSpec}.ObjectMeta.Labels:
	-: "map[serving.knative.dev/visibility:cluster-local]"
	+: "map[]"
```

Let me know if you see something obvious. Unless my cluster is not up to date. Digging more.

Reply:
The issue is, that the test uses the BYO feature for revision ("Bring your own" revision name). This means you need to to change the revision name to something unique on your own (like increasing a counter). Alternatively, you can use the auto revision name generation feature from Knative serving (which results in this random prefixes, like for pods of a deployment). You can servers side name generation with `-revision-name ""` (which might arguable not be the best option to use). Not sure anymore why we have settled on client side provided names by default.

However,  I would expect that `kn update` deals with percularity. Could you show me how you did the test ? Also, have you set the proper flag that the flag that you set also creates a new revision ? (look in the flags file how other do it)